tests/unit/data/validators/torch/test_image.py

```diff
@@ -0,0 +1,243 @@
+
+
+class TestImageValidator:
+    """Test ImageValidator."""
+
+    def setup_method(self) -> None:
+        """Set up the validator for each test method."""
+        self.validator = ImageValidator()
+
+    def test_validate_image_valid(self) -> None:
+        """Test validation of a valid image."""
+        image = torch.rand(3, 224, 224)
+        validated_image = self.validator.validate_image(image)
+        assert isinstance(validated_image, torch.Tensor)
+        assert validated_image.shape == (3, 224, 224)
+        assert validated_image.dtype == torch.float32
+
+    def test_validate_image_invalid_type(self) -> None:
+        """Test validation of an image with invalid type."""
+        with pytest.raises(TypeError, match="Image must be a torch.Tensor"):
+            self.validator.validate_image(np.random.default_rng().random((3, 224, 224)))
+
+    def test_validate_image_invalid_dimensions(self) -> None:
+        """Test validation of an image with invalid dimensions."""
+        with pytest.raises(ValueError, match="Image must have shape"):
+            self.validator.validate_image(torch.rand(224, 224))
+
+    def test_validate_image_invalid_channels(self) -> None:
+        """Test validation of an image with invalid number of channels."""
+        with pytest.raises(ValueError, match="Image must have 3 channels"):
+            self.validator.validate_image(torch.rand(1, 224, 224))
+
+    def test_validate_gt_label_valid(self) -> None:
+        """Test validation of a valid ground truth label."""
+        label = torch.tensor(1)
+        validated_label = self.validator.validate_gt_label(label)
+        assert isinstance(validated_label, torch.Tensor)
+        assert validated_label.dtype == torch.bool
+        assert validated_label.item() is True
+
+    def test_validate_gt_label_none(self) -> None:
+        """Test validation of a None ground truth label."""
+        assert self.validator.validate_gt_label(None) is None
+
+    def test_validate_gt_label_invalid_type(self) -> None:
+        """Test validation of a ground truth label with invalid type."""
+        with pytest.raises(TypeError, match="Ground truth label must be an integer or a torch.Tensor"):
+            self.validator.validate_gt_label("1")
+
+    def test_validate_gt_label_invalid_shape(self) -> None:
+        """Test validation of a ground truth label with invalid shape."""
+        with pytest.raises(ValueError, match="Ground truth label must be a scalar"):
+            self.validator.validate_gt_label(torch.tensor([0, 1]))
+
+    def test_validate_gt_mask_valid(self) -> None:
+        """Test validation of a valid ground truth mask."""
+        mask = torch.randint(0, 2, (1, 224, 224))
+        validated_mask = self.validator.validate_gt_mask(mask)
+        assert isinstance(validated_mask, Mask)
+        assert validated_mask.shape == (224, 224)
+        assert validated_mask.dtype == torch.bool
```

Comment:
Same. We convert int to bool. Not sure if the users will realise that they have to use the inputs returned from the validate method.